lib/crew_lockfile.rb

```diff
@@ -0,0 +1,568 @@
+                trace { "sleep <#{sleeptime}>..." }
+                sleep sleeptime
+              when false
+                trace { 'found invalid lock and removing' }
+                begin
+                  File.unlink @path
+                  @thief = true
+                  warn "<#{@path}> stolen by <#{Process.pid}> at <#{timestamp}>"
+                  trace { 'i am a thief!' }
+                rescue Errno::ENOENT => e
+                  warn(errmsg(e))
+                end
+                trace { "suspending <#{@suspend}>" }
+                sleep @suspend
+              when nil
+                raise MaxTriesLockError, "surpassed retries <#{@retries}>" if
+                  @retries && (n_retries >= @retries)
+              end
+              retry
+            end
+          end
+        rescue Timeout::Error
+          raise TimeoutLockError, "surpassed timeout <#{@timeout}>"
+        end
+
+        if block_given?
+          stolen = false
+          @refresher = (@refresh ? new_refresher : nil)
+          begin
+            begin
+              ret = yield @path
+            rescue StolenLockError
+              stolen = true
+              raise
+            end
+          ensure
+            begin
+              if @refresher&.status
+                begin
+                  @semaphore.synchronize do
+                    @refresher.kill
+                  end
+                rescue StandardError
+                  @refresher.kill
+                end
+              end
+              @refresher = nil
+            ensure
+              unlock unless stolen
+            end
+          end
+        else
+          @refresher = (@refresh ? new_refresher : nil)
+          ObjectSpace.define_finalizer self, @clean if @clean
+          ret = self
+        end
+      rescue Errno::ESTALE, Errno::EIO => e
+        raise(NFSLockError, errmsg(e))
+      end
+
+      return ret
+    end
+
+    def sweep
+      glob = File.join(@dirname, '.*lck')
+      paths = Dir[glob]
+      paths.each do |path|
+        basename = File.basename path
+        pat = /^\s*\.([^_]+)_([^_]+)/o
+        if pat.match(basename)
+          host = ::Regexp.last_match(1)
+          pid = ::Regexp.last_match(2)
+        else
+          next
+        end
+        host.gsub!(/^\.+|\.+$/, '')
+        quad = host.split '.'
+        host = quad.first
+        pat = /^\s*#{host}/i
+        if pat.match(HOSTNAME) && /^\s*\d+\s*$/.match(pid)
+          if alive?(pid)
+            trace { "process <#{pid}> on <#{host}> is still alive" }
+            trace { "ignoring <#{path}>" }
+          else
+            trace { "process <#{pid}> on <#{host}> is no longer alive" }
+            trace { "sweeping <#{path}>" }
+            FileUtils.rm_f path
+          end
+        else
+          trace { "ignoring <#{path}> generated by <#{host}>" }
+        end
+      rescue StandardError
+        next
+      end
+    rescue StandardError => e
+      warn(errmsg(e))
```

Comment:
Add `.lightred`?

Reply:
Added `.lightred` to the errmsg function.